microbench/src/test/java/io/netty/microbench/http2/Http2FrameWriterBenchmark.java

```diff
@@ -0,0 +1,491 @@
+        }
+
+        @Override
+        public Class<? extends ServerChannel> serverChannelClass() {
+            return NioServerSocketChannel.class;
+        }
+
+        @Override
+        public Class<? extends Channel> clientChannelClass() {
+            return NioSocketChannel.class;
+        }
+
+        @Override
+        public ByteBufAllocator clientAllocator() {
+            return PooledByteBufAllocator.DEFAULT;
+        }
+    };
+    private static final Environment[] environments = { boostrapEnv(NIO_POOLED_PARAMS),
+            boostrapEnv(NIO_UNPOOLED_PARAMS) };
+    private static final byte[][] dataPayload = { createData(256), createData(DEFAULT_MAX_FRAME_SIZE),
+            createData(MAX_FRAME_SIZE_UPPER_BOUND), createData(10 * MAX_FRAME_SIZE_UPPER_BOUND) };
+
+    @Param({ "0", "1", "2", "3" })
+    public int dataPayloadIndex;
+
+    @Param({ "0", "1" })
+    public int envIndex;
+
+    @Param({ "0", "255" })
+    public int padding;
+
+    @AfterClass
+    public static void teardown() {
+        for (Environment env : environments) {
+            try {
+                env.teardown();
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Benchmark
+    public void writeData() {
+        doWriteData(environments[envIndex], dataPayload[dataPayloadIndex], padding);
+    }
+
+    private static void doWriteData(Environment env, byte[] data, int padding) {
+        ByteBuf buf = env.clientContext.alloc().buffer(data.length);
+        buf.writeBytes(data);
+        env.clientWriter.writeData(env.clientContext, 3, buf, padding, true, env.clientContext.newPromise());
+    }
+
+    private static byte[] createData(int length) {
+        byte[] result = new byte[length];
+        new Random().nextBytes(result);
+        return result;
+    }
+
+    private static final class Environment {
+        ChannelHandlerContext clientContext;
+        Http2FrameWriter clientWriter;
+
+        void teardown() throws InterruptedException {
+        }
+    }
+
+    interface EnvironmentParameters {
+        EventLoopGroup newEventLoopGroup();
+
+        Class<? extends ServerChannel> serverChannelClass();
+
+        Class<? extends Channel> clientChannelClass();
+
+        ByteBufAllocator clientAllocator();
+    }
+
+    private static Environment boostrapEnv(final EnvironmentParameters params) {
+        final Environment env = new Environment();
+        env.clientWriter = new DefaultHttp2FrameWriter();
+        env.clientContext = new ChannelHandlerContext() {
+            EventExecutor executor = new DefaultEventExecutor();
+            EmbeddedChannel channel = new EmbeddedChannel();
+            SocketAddress localAddress;
+            ByteBufAllocator alloc = params.clientAllocator();
+            Http2Connection connection = new DefaultHttp2Connection(false);
+            Http2ConnectionHandler connectionHandler = new Http2ConnectionHandler(
```

Comment:
Do you think it would make sense to pull this out into a helper class for testing?